### ISSUE TITLE
Feature request: Arch Linux ARM/Manjaro ARM build support

### ISSUE THREAD

**Issue by Folaht** [SpottyMatt/raspbian-stepmania-build#3]
I tried it on Arch Linux ARM but get a
`Can not compile due to: #include <sys/io.h> file not found`
error

https://github.com/stepmania/stepmania/blob/5_1-new/src/arch/Lights/LightsDriver_LinuxParallel.cpp

**Comment by SpottyMatt** [comment]
1. Which Raspbery Pi?
2. Which Raspbian OS?
3. Is it a brand-new OS install, or have you used the Pi for other things, before?
4. Are you using the https://github.com/stepmania/stepmania/tree/f1ebe8d9d82b8272ddc2c63967780db28430e370 commit for StepMania, or have you changed to a new commit?
5. Please provide the complete output of the compile error message.

**Comment by Folaht** [comment]
1. Rpi4
2. No Raspbian, I prefer Manjaro on everything. Manjaro ARM Xfce4.
3. Brand new OS install
4. No.
5. 
```
[ 44%] Building CXX object src/CMakeFiles/StepMania.dir/arch/Lights/LightsDriver_LinuxParallel.cpp.o
/home/manietap/Jeux/raspbian-stepmania-build/stepmania/src/arch/Lights/LightsDriver_LinuxParallel.cpp:6:10: erreur fatale: sys/io.h : Aucun fichier ou dossier de ce type
   6 | #include <sys/io.h>
     |          ^~~~~~~~
compilation terminée.
```

I tried raspbian, there's no <sys/io.h> issue there.
It seems that raspbian has arm-linux-gnueabihf-gcc installed and 
arch not having that available for aarch64.

**Comment by SpottyMatt** [comment]
Per https://archlinuxarm.org/forum/viewtopic.php?t=14185&p=62799 , it _sounds_ like this is not something I can fix, but something [StepMania's own codebase](https://github.com/stepmania/stepmania) will have to fix.

That's... a long shot. There's no "official" ARM support to my knowledge. I can figure out packages to install and script up their installation, but if the core problem really reduces to 

> "StepMania compiles against a library that isn't available for Manjaro/ARCH Linux on ARM,"

then there's nothing really I can do - the software just isn't compatible with that distro setup.

Further, there's an old discussion of `sys/io.h` over here: https://bugzilla.redhat.com/show_bug.cgi?id=1116162 that sheds more light:

> The sys/ headers are architecture and OS dependent. They do not exist across all targets and io.h in particular is intended for very lowl-level non-portable uses often in coordination with the kernel. The only targets that provide sys/io.h are x86*, Alpha, IA64, and 32-bit ARM. No other systems provide it.

So it really does sound like _some_ platforms have had these low-level IO libraries produced for them, and StepMania uses them, and Arch does _not_ provide them for the Pi4B. In that case, the only solution is to refactor StepMania (unlikely) or for someone to start providing them for Arch on the Raspberry Pi (more likely, IMO, but perhaps not any time soon...)

... but if _you_ manage to figure it out... it may be possible to update these build scripts so they'll work as expected for the next person.

If I _had_ to work on this, I'd probably see if I could find the source code that provides them for Raspbian for the Pi4B, and see if I could get that to compile on Arch on the 4B. You may have to [shave a lot of yaks'](https://americanexpress.io/yak-shaving/) worth of dependencies to get that, and it might not end up possible.

My last Arch-on-RaspberryPi adventure - for the 2B - ended with a failure after such an attempt - I needed something FFMPEG or Audio-related, and the package simply didn't exist for Arch+Arm+RPi, and its dependencies' didn't exist, and the particular version I needed was so old that I couldn't find its source code to compile it myself, so I just couldn't install the package I needed and had to abandon Arch altogether and go back to Raspbian.

**TL;DR:** I'm happy to leave this issue open as an outstanding request, but I can't promise I'll become a Manjaro expert and figure out how to solve it for you.
